### PR TITLE
Utils Class Update

### PR DESCRIPTION
**Pull request by SeanCarrick**
Since we are attempting to build a professional, high-quality, integrated load tracking and accounting system for owner/operator truck drivers, I have updated the `com.pekinsoft.loadmaster.utils.Utils` class to help us achieve this end.

As we all know, each operating system has a standard for where on the system applications are supposed to store data, preferences, and log files. For this reason, I have added the following four (4) classes to the `Utils` class:

- `getApplicationFolderByOS`: checks the operating system name from the Java VM and returns a string of the path where that OS prefers applications to store their custom files.
- `getApplicationDataFolderByOS`: checks the operating system name from the Java VM and returns a string of the path where that OS prefers application to store their data files.
- `getApplicationSettingsLocationByOS`: checks the operating system name from the Java VM and returns a string of the path where that OS prefers applications to store their settings and configuration files.
- `getSystemLogLocationByOS`: checks the operating system name from the Java VM and returns a string of the path where that OS prefers applications to place their log files.

With this in mind, if we truly want a polished, professional application, we will need to update our code that sets up our folder structure and `AppLogger` class. Using these methods, we can place all of our Load Master™-generated files in the appropriate location for the system on which the user is running it.

@jkovalsky, if you approve of this, please just merge this branch to master. Otherwise, feel free to delete this branch and close the PR without merging. Thank you.

-SC
Signed-off-by:sean <sean@laptop.pekinsoft.com>
Signed-off-by:Sean Carrick <sean@pekinsoft.com>

### CODE REVIEW

File: Load Master/src/com/pekinsoft/loadmaster/utils/Utils.java [hunk 1]
```diff
@@ -62,6 +65,128 @@ private Utils () {
...
+     * this one has the same name, but this one **does not** end with a slash (/) 
+     * character. That is the only difference between the configuration file and  
+     * the application data folder.</dd></dl>
+     * 
+     * @param osName the name of the operating system upon which the application
+     *               is running.
+     * @return the OS-specific storage location.
+     */
+    public static String getApplicationFolderByOS(String osName) {
+        String location = "";
```
Comment (jkovalsky) [inline Load Master/src/com/pekinsoft/loadmaster/utils/Utils.java:93]: Method can be simplified to:

        String location = "";
        String userHome = System.getProperty("user.home");
        if (osName.toLowerCase().contains("win")) {
            location = "AppData\\LoadMaster";
        } else if (osName.toLowerCase().contains("mac")) {
            location = "Library/Preferences/LoadMaster";
        } else {
            location = ".loadmaster";
        }
        return userHome + File.separator + location + File.separator;

File: Load Master/src/com/pekinsoft/loadmaster/utils/Utils.java [hunk 2]
```diff
@@ -62,6 +65,128 @@ private Utils () {
...
+     * | Microsoft Windows™ | {user.home}\\AppData\\LoadMaster\\data\\ |
+     * | Apple Mac OS-X™ | {user.home}/Library/Application Data/LoadMaster/data/ |
+     * | Linux and Solaris™ | {user.home}/.loadmaster/data/ |
+     * 
+     * @param osName the name of the operating system upon which the application
+     *               is running.
+     * @return the OS-specific storage location.
+     */
+    public static String getApplicationDataFolderByOS(String osName) {
+        String location = "";
```
Comment (jkovalsky) [inline Load Master/src/com/pekinsoft/loadmaster/utils/Utils.java:125]: Using my suggestion above this method can be simplified to:

        String applicationFolder = getApplicationFolderByOS(osName);
        if (osName.toLowerCase().contains("mac")) {
            applicationFolder = System.getProperty("user.home") + "/Library/Application Data/LoadMaster/";
        }
        return applicationFolder + "data" + File.separator;

File: Load Master/src/com/pekinsoft/loadmaster/utils/Utils.java [hunk 3]
```diff
@@ -62,6 +65,128 @@ private Utils () {
...
+     * this one has the same name, but this one **does not** end with a slash (/) 
+     * character. That is the only difference between the configuration file and  
+     * the application data folder.</dd></dl>
+     * 
+     * @param osName the name of the operating system upon which the application
+     *               is running.
+     * @return the OS-specific storage location.
+     */
+    public static String getApplicationSettingsLocationByOS(String osName) {
+        String location = "";
```
Comment (jkovalsky) [inline Load Master/src/com/pekinsoft/loadmaster/utils/Utils.java:162]: Using my suggestion above this method can be simplified to:

        String settingsLocationFolder = getApplicationFolderByOS(osName);
        if (osName.toLowerCase().contains("win")) {
            settingsLocationFolder = settingsLocationFolder + "LoadMaster.cnf";
        } else if (osName.toLowerCase().contains("mac")) {
            settingsLocationFolder = settingsLocationFolder + "LoadMaster.pref";
        } else {
            settingsLocationFolder = settingsLocationFolder + "loadmaster.conf";
        }
        return settingsLocationFolder;

**Review by jkovalsky — APPROVED** [review]
I suggest to either incorporate my simplifications or rather go with pull request #66 .

**Review by jkovalsky — CHANGES_REQUESTED** [review]
Please consider my suggestions or at least reply to my comments. Thank you.

**Comment by SeanCarrick** [comment]
I am just closing this PR since it has failed all Travis checks. I will redo this PR taking your suggestions and possibly improving upon them.

-SC

**Pull request closed without merging**